demo/steady/demo.py

```diff
@@ -0,0 +1,108 @@
+#!/bin/python3
+
+import os
+import time
+import requests
+import datetime
+import argparse
+
+# File locations
+SCRIPT_DIR = os.path.join(os.path.dirname(__file__))
+EXAMPLE_SQL = os.path.join(SCRIPT_DIR, "program.sql")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--api-url", required=True,
+                        help="Feldera REST API URL (e.g., http://localhost:8080 or https://sandbox-staging.feldera.com")
+    parser.add_argument('--num-total', required=False, default=10000000000000000000000000)
+    parser.add_argument('--target-rate-per-second', required=True)
+    parser.add_argument("--with-bearer-token", required=False,
+                        help="Authorization Bearer token")
+    args = parser.parse_args()
+    if args.with_bearer_token is None:
+        headers = {}
+    else:
+        headers = {
+            "authorization": f"Bearer {args.with_bearer_token}"
+        }
+
+    # Feldera REST API URL
+    api_url = args.api_url
+    print(f"Feldera REST API URL: {api_url}")
+
+    # Total
+    num_total = int(args.num_total)
+    print(f"Total number: {num_total}")
+
+    # Target rate
+    target_rate_per_second = int(args.target_rate_per_second)
+    print(f"Target rate per second: {target_rate_per_second}")
+
+    # Test connectivity by fetching existing programs
+    requests.get(f"{api_url}/v0/programs", headers=headers).raise_for_status()
+
+    # Create program
+    program_name = "demo-steady-program"
+    program_sql = open(EXAMPLE_SQL).read()
+    response = requests.put(f"{api_url}/v0/programs/{program_name}", headers=headers, json={
+        "description": "",
+        "code": program_sql
+    })
+    response.raise_for_status()
+    program_version = response.json()["version"]
+
+    # Compile program
+    print(f"Compiling program {program_name} (version: {program_version})...")
+    requests.post(f"{api_url}/v0/programs/{program_name}/compile", headers=headers, json={"version": program_version}).raise_for_status()
+    while True:
```

Comment:
this should be a utility function in some library

Reply:
(see other comment)